docs/internal/continuous-profiler.md

```diff
@@ -0,0 +1,255 @@
+# Continuous profiler
+
+> [!IMPORTANT]  
+> Continuous profiler is an experimental feature. It will be subject to change,
+> when <https://github.com/open-telemetry/oteps/pull/239> or <https://github.com/open-telemetry/oteps/pull/237>
+> will be ready and merged.
+
+## Thread sampling
+
+Thread sampling can be enabled by the custom plugin.
+Plugin is responsible for parsing dense, data format and exporting it in
+the appropriate format.
+
+### How does the thread sampler work?
+
+The profiler leverages [.NET profiling](https://docs.microsoft.com/en-us/dotnet/framework/unmanaged-api/profiling/)
+to perform periodic call stack sampling. For every sampling period, the runtime
+ suspended and the samples for all managed thread are saved into the buffer,
+ then the runtime resumes.
+
+The separate managed-thread is processing data from the buffer and export it
+the way defined by the plugin.
+
+To make the process more efficient, the sampler uses two independent buffers to
+store samples alternatively.
```

Comment:
```suggestion
The profiler uses the
[.NET profiler](https://docs.microsoft.com/en-us/dotnet/framework/unmanaged-api/profiling/)
to perform periodic call stack sampling. For every sampling period, the runtime
suspends execution and the samples for all managed thread are saved in the buffer,
then the runtime resumes.

The separate managed thread processes data from the buffer and exports it
in the format defined by the plugin. To make the process more efficient, the
sampler uses two independent buffers to store samples alternatively.
```

How long does runtime suspension last? Does it affect application performance significantly?

Reply:
Based on logs I have checked, for simple application it is ~150microseconds. It is vary based on the application size, machine, etc.